lib/src/revset.rs

```diff
@@ -121,16 +121,22 @@ pub trait RevsetFilterExtension: std::fmt::Debug + Any {
     fn matches_commit(&self, commit: &Commit) -> bool;
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum SignatureField {
+    Name,
+    Email,
+}
+
 #[derive(Clone, Debug)]
 pub enum RevsetFilterPredicate {
     /// Commits with number of parents in the range.
     ParentCount(Range<u32>),
     /// Commits with description matching the pattern.
     Description(StringPattern),
-    /// Commits with author name or email matching the pattern.
-    Author(StringPattern),
-    /// Commits with committer name or email matching the pattern.
-    Committer(StringPattern),
+    /// Commits with author field matching the pattern.
+    Author(SignatureField, StringPattern),
+    /// Commits with committer field matching the pattern.
+    Committer(SignatureField, StringPattern),
```

Comment:
Maybe these enum variants can be simply `AuthorName(_)`/`AuthorEmail(_)`/..?

We're going to introduce `AuthorDate(_)`, etc., so `Author<Thing>` seems more consistent. And we'll probably add `author_name()`/`_email()` revset function if needed.